### PR TITLE
provide a undefined/null speed limit when user provide empty field

### PR DESCRIPTION
**Pull request by kmer2016**
fix the first problem of [this issue](https://github.com/osrd-project/osrd/issues/6407)

The problem was that the way we converted the values entered in Speed limit field returned 0 when the field was left empty.

How to test ?
- On Infra Editor, load an infra
- Open Speed limit tool (image
- add some Speed limit with empty key/value
- save and check on network tab of devtools that the payload of the request POST on url http://localhost:4000/api/infra/4/ replace empty speed limit by undefined/null

### CODE REVIEW

**Review by Akctarus — APPROVED** [review]
Tested, LGTM

File: front/src/applications/editor/tools/rangeEdition/speedSection/SpeedSectionMetadataForm.tsx [hunk 1]
```diff
@@ -74,8 +74,8 @@ const SpeedSectionMetadataForm: FC = () => {
             {t('Editor.tools.speed-edition.additional-speed-limit')}
           </div>
         )}
-        {map(entity.properties.speed_limit_by_tag || {}, (value, key) => (
-          <div key={key} className="form-group field field-string">
+        {Object.entries(entity.properties.speed_limit_by_tag || {}).map(([key, value], index) => (
+          <div key={index} className="form-group field field-string">
```
Comment (Yohh) [inline front/src/applications/editor/tools/rangeEdition/speedSection/SpeedSectionMetadataForm.tsx:78]: can you use anything but the index in the key?

**Review by Yohh — CHANGES_REQUESTED** [review]

**Review by flomonster — APPROVED** [review]
Tested and it works :tada:.

File: front/src/applications/editor/tools/rangeEdition/speedSection/SpeedSectionMetadataForm.tsx (hunk 1 above)
Comment (Yohh) [reply front/src/applications/editor/tools/rangeEdition/speedSection/SpeedSectionMetadataForm.tsx:78]: ok for fixing it in another pr

**Review by Yohh — APPROVED** [review]

**Review by SharglutDev — APPROVED** [review]
Lgtm and tested. As specified by @Yohh, the key issue demands more work and will be fixed in a different [PR](https://github.com/osrd-project/osrd/pull/6461)

**Pull request merged**